taskt/Core/Automation/Commands/Error Handling Commands/BeginRetryCommand.cs

```diff
@@ -0,0 +1,235 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Threading;
+using System.Windows.Forms;
+using System.Xml.Serialization;
+using taskt.Core.Automation.Attributes.ClassAttributes;
+using taskt.Core.Automation.Attributes.PropertyAttributes;
+using taskt.Core.Automation.Engine;
+using taskt.Core.Script;
+using taskt.Core.Utilities.CommonUtilities;
+using taskt.UI.CustomControls;
+using taskt.UI.Forms;
+
+namespace taskt.Core.Automation.Commands
+{
+    [Serializable]
+    [Group("Error Handling Commands")]
+    [Description("This command defines a retry block which will retry the contained commands as long as the condition is not met or " +
+        "an error is thrown.")]
+    public class BeginRetryCommand : ScriptCommand
+    {
+        [XmlAttribute]
+        [PropertyDescription("Number of Retries")]
+        [InputSpecification("Enter or provide the number of retries.")]
+        [SampleUsage("3 || {vRetryCount}")]
+        [Remarks("")]
+        [PropertyUIHelper(PropertyUIHelper.UIAdditionalHelperType.ShowVariableHelper)]
+        public string v_RetryCount { get; set; }
+
+        [XmlAttribute]
+        [PropertyDescription("Retry Interval")]
+        [InputSpecification("Enter or provide the amount of time (in seconds) between each retry.")]
+        [SampleUsage("5 || {vRetryInterval}")]
+        [Remarks("")]
+        [PropertyUIHelper(PropertyUIHelper.UIAdditionalHelperType.ShowVariableHelper)]
+        public string v_RetryInterval { get; set; }
+
+        [XmlElement]
+        [PropertyDescription("Condition")]
+        [InputSpecification("Add a condition.")]
+        [SampleUsage("")]
+        [Remarks("Items in the retry scope will be executed, if the condition doesn't satisfy.")]
+        [PropertyUIHelper(PropertyUIHelper.UIAdditionalHelperType.ShowIfBuilder)]
+        public DataTable v_IfConditionsTable { get; set; }
```

Comment:
Was it your intention to disable the ability to add multiple if conditions despite using a ShowIfBuilder with a table? Wouldn't it be better to just replicate the drop down that BeginIf uses instead? I think you should either make it possible to add multiple conditions or remove the table entirely and replace it with a drop down if you're just going to use one.

Reply:
Agreed. I assumed given the UI I could add multiple conditions. A single condition should be sufficient for now.

Reply:
Added ability to add multiple if conditions.